configuration/cjdns.md

```diff
@@ -0,0 +1,229 @@
+
+## Setup of a Cjdns ProxyVM
+While you could install the Cjdns routing software in any AppVM, the benefit of a ProxyVM is that you can provide Cjdns and Hyperboria connectivity transparently to any AppVM without additional installation.
+
+##### Limitations:
+- The Cjdns-ProxyVM acts as a NAT-Gateway, so the Cjdns traffic between the ProxyVM and an AppVM is not encrypted. However, since this traffic is only within Qubes, it's probably acceptable for you. See https://github.com/cjdelisle/cjdns/blob/master/doc/nat-gateway.md for details.
+- Since the Cjdns-ProxyVM itself is also behind a NAT (of `sys-firewall` or `sys-net` VM), the Cjdns router won't connect to other Cjdns routers in your local network `automagically` (for details regarding `beacon mode` see: https://github.com/cjdelisle/cjdns/blob/master/doc/configure.md). So no 'zero-configuration' unfortunately and some additional `manual configuration` is required. However, not everybody needs this anyway and if so, this needs to be done only once for the ProxyVM and not every AppVM. 
+
+### 1. Create a TemplateVM (cjdns-gw)
+We'll build Cjdns from the GitHub sources, which requires some additional packages. Best is if you `Clone` any of the default TemplateVMs that come with Qubes 4.0. This How-To uses `Debian 9 (stretch)` template to build Cjdns and run the ProxyVM.
+
+So `clone` the `debian-9` TemplateVM via the Qubes Manager and assign the name **cjdns-gw**. Why? The Whonix TemplateVM (whonix-gw) has the same naming, so let's keep some consistency.
+
+Note: You might want to change the `Networking` of the `cjdns-gw` TemplateVM to `sys-whonix`.
+
+### 2. Install required software in the TemplateVM (cjdns-gw)
+The [Cjdns installation instructions for Debian jessie](https://github.com/cjdelisle/cjdns/blob/master/doc/install/debian-jessie.md) work for *stretch (Debian 9)* as well:
+
+Launch the `cjdns-gw` TemplateVM and allow outbound network connections for 5 minutes via `Edit qube firewall rules` in the Qubes Manager.
+
+Install the required packages to build Cjdns:
+```
+user@cjdns-gw:~$ sudo apt-get install nodejs git build-essential
+```
+### 3. Checkout and build Cjdns
+This can be done in the home directory of the `user` in the `cjdns-gw` TemplateVM.
+```
+user@cjdns-gw:~$ cd
+user@cjdns-gw:~$ pwd
+/home/user
+user@cjdns-gw:~$ git clone https://github.com/cjdelisle/cjdns.git
+...
+user@cjdns-gw$ cd cjdns
+user@cjdns-gw:~/cjdns$ ./do
+...
+```
+
+### 4. Install Cjdns as a system service
+If the previous step completed without errors, install Cjdns as as system service (`systemd`).
+
+While you're still in the Cjdns directory, type:
+```
+user@cjdns-gw:~/cjdns$ sudo cp cjdroute /usr/bin
+user@cjdns-gw:~/cjdns$ sudo cp contrib/systemd/cjdns*.service /etc/systemd/system/
```

Comment:
It might be a good idea to add `qvm-service` integration here, to avoid starting the service in unwanted VMs. It's just one line, see https://www.qubes-os.org/doc/qubes-service/